datalad_registry_client/opts.py

```diff
@@ -64,17 +66,23 @@ def process_args(
         if sibling not in remotes:
             raise ValueError("Unknown sibling: {}".format(sibling))
 
-    if not url:
-        if not sibling:
-            raise ValueError(
-                "Must specify URL to use when sibling isn't given")
+    urls: List[str] = []
+    if url:
+        urls.append(url)
+    elif sibling:
         url = repo.config.get("remote.{}.url".format(sibling))
-        if not url:
+        if url is None:
             raise ValueError("Could not find URL for {}".format(sibling))
+        urls.append(url)
+    else:
+        for r in repo.get_remotes(with_urls_only=True):
+            url = repo.get_remote_url(r)
+            assert url is not None
+            urls.append(url)
 
     endpoint = endpoint or repo.config.get(
         "datalad_registry.endpoint",
         "http://127.0.0.1:5000/v1")
     return dict(ds=ds, ds_id=ds_id,
-                sibling=sibling, url=url, url_encoded=url_encode(url),
+                sibling=sibling, urls=[(url, url_encode(url)) for url in urls],
```

Comment:
but why to do it `url_encode(url)` in this function at all and not outside whenever `url_encode`ing is actually needed, like [here](https://github.com/datalad/datalad-registry/pull/46/files#diff-e8ba59201fb7b17dd59ceb249e21c397fe879dce205a093ff03df88e13921fb8R51)?  outside code might decide to handle those urls differently so better that special handling is done where/when needed.
sorry if I am missing something obvious

Reply:
That's just the way the code was when I found it.  However, encoding the URL in `opts.process_args()` does lead to the encoded URL being included in the result dict, which seems undesirable, so I'll change it.

Reply:
@yarikoptic On a related note, should the result dicts include everything that `opts.process_args()` returns?  Including `urls` at the very least seems odd, as it would probably be a better idea for each individual result dict to have a single `url` key for respective URL, and then there's the `ds_id` key....

Reply:
I agree -- a separate record per url would be better, since could have different status etc.  I would also add `url_encoded` as well in that result entry though to simplify possible troubleshooting etc.